### DISCUSSION TITLE
why YourTTS's vits not used speaker_manager.encoder in training forward?

### DISCUSSION THREAD

**Discussion by YooSungHyun** [coqui-ai/TTS#2579] [General Q&A]
i want to `"use_speaker_encoder_as_loss": true` then, what kind of speaker embedding is had to used?

d_vector? or speaker_embedding?

why speaker_manager.encoder's forward output is not used in vits's speaker_embedding?
I can see that is just use to calc `pred_embs` for loss backpropagate.

so, i think i have to make my task process just like this
1. I have to load speaker_encoder first
2. make d_vector or something
3. `"use_speaker_encoder_as_loss": true` and `"use_d_vector_file": true`
4. fine tuning speaker encoder to SCL and finetuning vits
5. use fine-tuned speaker encoder for inference
6. make new d_vector for another fine-tune speaker encoder
7. if when i want to another fine tune speaker_encoder, i have to use new d_vector and just backprob to speaker_manager.encoder
is that right?

how about this?
  1) load speaker_encoder first
  2) `"use_speaker_encoder_as_loss": true` then just use `speaker_embedding` made by `speaker_manager.encoder.eval()` and `forward()`
  3) fine tuning all network by calced SCL and another loss and backprob
  4)  use fine-tuned speaker encoder for inference
  5) if when i want to another fine tune speaker_encoder, just do it first step to here
is this wrong?

very curious 🤔

**Comment by YooSungHyun** [comment]
maybe. that is why some kind of `many speakers's different audio length`'s padding issue... so, can not sampling suitable speaker's audio problem...? (for make speaker_embedding made by `speaker_manager.encoder.forward()`)
